ci/conan/all/conanfile.py

```diff
@@ -136,283 +117,147 @@ class ArrowConan(ConanFile):
-
-    def _with_glog(self, required=False):
-        if required or self.options.with_glog == "auto":
-            return False
-        else:
-            return bool(self.options.with_glog)
-
-    def _with_grpc(self, required=False):
-        if required or self.options.with_grpc == "auto":
-            return self._with_flight_rpc()
-        else:
-            return bool(self.options.with_grpc)
-
-    def _with_boost(self, required=False):
-        if required or self.options.with_boost == "auto":
-            if self.options.gandiva:
-                return True
-            version = Version(self.version)
-            if version.major == "1":
-                if self._parquet() and self.settings.compiler == "gcc" and self.settings.compiler.version < Version("4.9"):
-                    return True
-            elif version.major >= "2":
-                if is_msvc(self):
-                    return True
-            return False
-        else:
-            return bool(self.options.with_boost)
-
-    def _with_thrift(self, required=False):
-        # No self.options.with_thrift exists
-        return bool(required or self._parquet())
-
-    def _with_utf8proc(self, required=False):
-        if required or self.options.with_utf8proc == "auto":
-            return bool(self._compute() or self.options.gandiva)
-        else:
-            return bool(self.options.with_utf8proc)
-
-    def _with_llvm(self, required=False):
-        if required or self.options.with_llvm == "auto":
-            return bool(self.options.gandiva)
-        else:
-            return bool(self.options.with_llvm)
-
-    def _with_openssl(self, required=False):
-        if required or self.options.with_openssl == "auto":
-            return bool(self.options.encryption or self._with_flight_rpc() or self.options.with_s3)
-        else:
-            return bool(self.options.with_openssl)
-
-    def _with_rapidjson(self):
-        if self.options.with_json:
-            return True
-        if Version(self.version) >= "7.0.0" and self.options.encryption:
-            return True
-        return False
+    def _requires_rapidjson(self):
+        return self.options.with_json or self.options.encryption
 
     def requirements(self):
-        if self._with_thrift():
-            self.requires("zlib/1.2.13")
+        if self.options.with_thrift:
             self.requires("thrift/0.17.0")
-        if self._with_protobuf():
-            self.requires("protobuf/3.21.4")
-        if self._with_jemalloc():
+        if self.options.with_protobuf:
+            self.requires("protobuf/3.21.9")
+        if self.options.with_jemalloc:
             self.requires("jemalloc/5.3.0")
         if self.options.with_mimalloc:
             self.requires("mimalloc/1.7.6")
-        if self._with_boost():
-            self.requires("boost/1.80.0")
-        if self._with_gflags():
+        if self.options.with_boost:
+            self.requires("boost/1.84.0")
```

Comment:
It seems that the boost version is too high for us.

```
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:1821 ] _boost_PATH_SUFFIXES = "boost-1_81_0;boost_1_81_0;boost/boost-1_81_0;boost/boost_1_81_0;boost-1_81;boost_1_81;boost/boost-1_81;boost/boost_1_81;boost-1_80_0;boost_1_80_0;boost/boost-1_80_0;boost/boost_1_80_0;boost-1_80;boost_1_80;boost/boost-1_80;boost/boost_1_80;boost-1_79_0;boost_1_79_0;boost/boost-1_79_0;boost/boost_1_79_0;boost-1_79;boost_1_79;boost/boost-1_79;boost/boost_1_79;boost-1_78_0;boost_1_78_0;boost/boost-1_78_0;boost/boost_1_78_0;boost-1_78;boost_1_78;boost/boost-1_78;boost/boost_1_78;boost-1_77_0;boost_1_77_0;boost/boost-1_77_0;boost/boost_1_77_0;boost-1_77;boost_1_77;boost/boost-1_77;boost/boost_1_77;boost-1_76_0;boost_1_76_0;boost/boost-1_76_0;boost/boost_1_76_0;boost-1_76;boost_1_76;boost/boost-1_76;boost/boost_1_76;boost-1_75_0;boost_1_75_0;boost/boost-1_75_0;boost/boost_1_75_0;boost-1_75;boost_1_75;boost/boost-1_75;boost/boost_1_75;boost-1_74_0;boost_1_74_0;boost/boost-1_74_0;boost/boost_1_74_0;boost-1_74;boost_1_74;boost/boost-1_74;boost/boost_1_74;boost-1_73_0;boost_1_73_0;boost/boost-1_73_0;boost/boost_1_73_0;boost-1_73;boost_1_73;boost/boost-1_73;boost/boost_1_73;boost-1_72_0;boost_1_72_0;boost/boost-1_72_0;boost/boost_1_72_0;boost-1_72;boost_1_72;boost/boost-1_72;boost/boost_1_72;boost-1_71_0;boost_1_71_0;boost/boost-1_71_0;boost/boost_1_71_0;boost-1_71;boost_1_71;boost/boost-1_71;boost/boost_1_71;boost-1_70_0;boost_1_70_0;boost/boost-1_70_0;boost/boost_1_70_0;boost-1_70;boost_1_70;boost/boost-1_70;boost/boost_1_70;boost-1_69_0;boost_1_69_0;boost/boost-1_69_0;boost/boost_1_69_0;boost-1_69;boost_1_69;boost/boost-1_69;boost/boost_1_69;boost-1_68_0;boost_1_68_0;boost/boost-1_68_0;boost/boost_1_68_0;boost-1_68;boost_1_68;boost/boost-1_68;boost/boost_1_68;boost-1_67_0;boost_1_67_0;boost/boost-1_67_0;boost/boost_1_67_0;boost-1_67;boost_1_67;boost/boost-1_67;boost/boost_1_67;boost-1_66_0;boost_1_66_0;boost/boost-1_66_0;boost/boost_1_66_0;boost-1_66;boost_1_66;boost/boost-1_66;boost/boost_1_66;boost-1_65_0;boost_1_65_0;boost/boost-1_65_0;boost/boost_1_65_0;boost-1_65;boost_1_65;boost/boost-1_65;boost/boost_1_65;boost-1_64_0;boost_1_64_0;boost/boost-1_64_0;boost/boost_1_64_0;boost-1_64;boost_1_64;boost/boost-1_64;boost/boost_1_64;boost-1_63_0;boost_1_63_0;boost/boost-1_63_0;boost/boost_1_63_0;boost-1_63;boost_1_63;boost/boost-1_63;boost/boost_1_63;boost-1_62_0;boost_1_62_0;boost/boost-1_62_0;boost/boost_1_62_0;boost-1_61;boost_1_61;boost/boost-1_61;boost/boost_1_61;boost-1_61_0;boost_1_61_0;boost/boost-1_61_0;boost/boost_1_61_0;boost-1_62;boost_1_62;boost/boost-1_62;boost/boost_1_62;boost-1_60_0;boost_1_60_0;boost/boost-1_60_0;boost/boost_1_60_0;boost-1_60;boost_1_60;boost/boost-1_60;boost/boost_1_60;boost-1_84_0;boost_1_84_0;boost/boost-1_84_0;boost/boost_1_84_0;boost-1_84;boost_1_84;boost/boost-1_84;boost/boost_1_84;boost-1_83_0;boost_1_83_0;boost/boost-1_83_0;boost/boost_1_83_0;boost-1_83;boost_1_83;boost/boost-1_83;boost/boost_1_83;boost-1_82_0;boost_1_82_0;boost/boost-1_82_0;boost/boost_1_82_0;boost-1_82;boost_1_82;boost/boost-1_82;boost/boost_1_82;boost-1_81_0;boost_1_81_0;boost/boost-1_81_0;boost/boost_1_81_0;boost-1_81;boost_1_81;boost/boost-1_81;boost/boost_1_81;boost-1_80_0;boost_1_80_0;boost/boost-1_80_0;boost/boost_1_80_0;boost-1_80;boost_1_80;boost/boost-1_80;boost/boost_1_80;boost-1_79_0;boost_1_79_0;boost/boost-1_79_0;boost/boost_1_79_0;boost-1_79;boost_1_79;boost/boost-1_79;boost/boost_1_79;boost-1_78_0;boost_1_78_0;boost/boost-1_78_0;boost/boost_1_78_0;boost-1_78;boost_1_78;boost/boost-1_78;boost/boost_1_78;boost-1_77_0;boost_1_77_0;boost/boost-1_77_0;boost/boost_1_77_0;boost-1_77;boost_1_77;boost/boost-1_77;boost/boost_1_77;boost-1_76_0;boost_1_76_0;boost/boost-1_76_0;boost/boost_1_76_0;boost-1_76;boost_1_76;boost/boost-1_76;boost/boost_1_76;boost-1_75_0;boost_1_75_0;boost/boost-1_75_0;boost/boost_1_75_0;boost-1_75;boost_1_75;boost/boost-1_75;boost/boost_1_75;boost-1_74_0;boost_1_74_0;boost/boost-1_74_0;boost/boost_1_74_0;boost-1_74;boost_1_74;boost/boost-1_74;boost/boost_1_74;boost-1_73_0;boost_1_73_0;boost/boost-1_73_0;boost/boost_1_73_0;boost-1_73;boost_1_73;boost/boost-1_73;boost/boost_1_73;boost-1_72_0;boost_1_72_0;boost/boost-1_72_0;boost/boost_1_72_0;boost-1_72;boost_1_72;boost/boost-1_72;boost/boost_1_72;boost-1_71_0;boost_1_71_0;boost/boost-1_71_0;boost/boost_1_71_0;boost-1_71;boost_1_71;boost/boost-1_71;boost/boost_1_71;boost-1_70_0;boost_1_70_0;boost/boost-1_70_0;boost/boost_1_70_0;boost-1_70;boost_1_70;boost/boost-1_70;boost/boost_1_70;boost-1_69_0;boost_1_69_0;boost/boost-1_69_0;boost/boost_1_69_0;boost-1_69;boost_1_69;boost/boost-1_69;boost/boost_1_69;boost-1_68_0;boost_1_68_0;boost/boost-1_68_0;boost/boost_1_68_0;boost-1_68;boost_1_68;boost/boost-1_68;boost/boost_1_68;boost-1_67_0;boost_1_67_0;boost/boost-1_67_0;boost/boost_1_67_0;boost-1_67;boost_1_67;boost/boost-1_67;boost/boost_1_67;boost-1_66_0;boost_1_66_0;boost/boost-1_66_0;boost/boost_1_66_0;boost-1_66;boost_1_66;boost/boost-1_66;boost/boost_1_66;boost-1_65_1;boost_1_65_1;boost/boost-1_65_1;boost/boost_1_65_1;boost-1_65_0;boost_1_65_0;boost/boost-1_65_0;boost/boost_1_65_0;boost-1_65;boost_1_65;boost/boost-1_65;boost/boost_1_65;boost-1_64_0;boost_1_64_0;boost/boost-1_64_0;boost/boost_1_64_0;boost-1_64;boost_1_64;boost/boost-1_64;boost/boost_1_64;boost-1_63_0;boost_1_63_0;boost/boost-1_63_0;boost/boost_1_63_0;boost-1_63;boost_1_63;boost/boost-1_63;boost/boost_1_63;boost-1_62_0;boost_1_62_0;boost/boost-1_62_0;boost/boost_1_62_0;boost-1_62;boost_1_62;boost/boost-1_62;boost/boost_1_62;boost-1_61_0;boost_1_61_0;boost/boost-1_61_0;boost/boost_1_61_0;boost-1_61;boost_1_61;boost/boost-1_61;boost/boost_1_61;boost-1_60_0;boost_1_60_0;boost/boost-1_60_0;boost/boost_1_60_0;boost-1_60;boost_1_60;boost/boost-1_60;boost/boost_1_60;boost-1_59_0;boost_1_59_0;boost/boost-1_59_0;boost/boost_1_59_0;boost-1_59;boost_1_59;boost/boost-1_59;boost/boost_1_59;boost-1_58_0;boost_1_58_0;boost/boost-1_58_0;boost/boost_1_58_0;boost-1_58;boost_1_58;boost/boost-1_58;boost/boost_1_58"
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:1906 ] Boost_LIB_PREFIX = ""
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:1907 ] Boost_NAMESPACE = "boost"
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:953 ] _boost_COMPILER = "-gcc" (guessed)
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:1943 ] _boost_MULTITHREADED = "-mt"
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2021 ] _boost_ARCHITECTURE_TAG = "" (detected)
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2025 ] _boost_RELEASE_ABI_TAG = "-"
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2026 ] _boost_DEBUG_ABI_TAG = "-d"
-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2086 ] _boost_LIBRARY_SEARCH_DIRS_RELEASE = "Boost_INCLUDE_DIR-NOTFOUND/lib;Boost_INCLUDE_DIR-NOTFOUND/../lib;Boost_INCLUDE_DIR-NOTFOUND/stage/lib;PATHS;C:/boost/lib;C:/boost;/sw/local/lib"
CMake Error at /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR) (Required is at least
  version "1.58")
Call Stack (most recent call first):
  /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2393 (find_package_handle_standard_args)
  cmake_modules/ThirdpartyToolchain.cmake:310 (find_package)
  cmake_modules/ThirdpartyToolchain.cmake:1288 (resolve_dependency)
  CMakeLists.txt:543 (include)


-- [ /home/conan/.conan/data/cmake/3.29.0/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/share/cmake-3.29/Modules/FindBoost.cmake:2087 ] _boost_LIBRARY_SEARCH_DIRS_DEBUG = "Boost_INCLUDE_DIR-NOTFOUND/lib;Boost_INCLUDE_DIR-NOTFOUND/../lib;Boost_INCLUDE_DIR-NOTFOUND/stage/lib;PATHS;C:/boost/lib;C:/boost;/sw/local/lib"
```

Reply:
Yes!

Reply:
ARROW_PARQUET -> ARROW_WITH_THRIFT -> ARROW_USE_BOOST

Reply:
```
if(ARROW_PARQUET)
  set(ARROW_WITH_THRIFT ON)
endif()
```

```
if(ARROW_BUILD_INTEGRATION
   OR ARROW_BUILD_TESTS
   OR (ARROW_FLIGHT AND ARROW_BUILD_BENCHMARKS)
   OR (ARROW_S3 AND ARROW_BUILD_BENCHMARKS))
  set(ARROW_USE_BOOST TRUE)
  set(ARROW_BOOST_REQUIRE_LIBRARY TRUE)
elseif(ARROW_GANDIVA
       OR ARROW_WITH_THRIFT
       OR (NOT ARROW_USE_NATIVE_INT128))
  set(ARROW_USE_BOOST TRUE)
  set(ARROW_BOOST_REQUIRE_LIBRARY FALSE)
else()
  set(ARROW_USE_BOOST FALSE)
endif()
```